draco/analysis/transform.py

```diff
@@ -155,6 +136,45 @@ def setup(self, tel):
         )
         self.bt_rev["conjugate"] = np.where(feedmask, self.telescope.feedconj[triu], 0)
 
+
+class CollateProducts(TelescopeStreamMixIn, task.SingleTask):
+    """Extract and order the correlation products for map-making.
+
+    The task will take a sidereal task and format the products that are needed
+    or the map-making. It uses a BeamTransfer instance to figure out what these
+    products are, and how they should be ordered. It similarly selects only the
+    required frequencies.
+
+    It is important to note that while the input
+    :class:`~containers.SiderealStream` can contain more feeds and frequencies
+    than are contained in the BeamTransfers, the converse is not true. That is,
+    all the frequencies and feeds that are in the BeamTransfers must be found in
+    the timestream object.
+
+    Parameters
+    ----------
+    weight : string ('natural', 'uniform', or 'inverse_variance')
+        How to weight the redundant baselines when stacking:
+            'natural' - each baseline weighted by its redundancy (default)
+            'uniform' - each baseline given equal weight
+            'inverse_variance' - each baseline weighted by the weight attribute
+    """
+
+    weight = config.Property(proptype=str, default="natural")
```

Comment:
We may as well change this to an `enum` and not bother removing the check in `setup` - that way an error could get caught by `caput-pipeline lint`

Reply:
Oh yeah, good point.